packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -142,29 +150,38 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
    */
   public readonly artifactBucket: s3.IBucket;
 
+  private readonly managedBucket: boolean;
   private readonly stages = new Array<Stage>();
   private eventsRole?: iam.Role;
   private readonly pipelineResource: CfnPipeline;
+  private readonly crossAccountActionRoles: Map<cdk.Stack, iam.Role>;
   private readonly crossRegionReplicationBuckets: { [region: string]: string };
   private readonly artifactStores: { [region: string]: any };
   private readonly _crossRegionScaffoldStacks: { [region: string]: CrossRegionScaffoldStack } = {};
 
-  constructor(scope: cdk.Construct, id: string, props?: PipelineProps) {
+  constructor(scope: cdk.Construct, id: string, props: PipelineProps = {}) {
     super(scope, id);
-    props = props || {};
 
     cpapi.validateName('Pipeline', props.pipelineName);
 
     // If a bucket has been provided, use it - otherwise, create a bucket.
     let propsBucket = props.artifactBucket;
     if (!propsBucket) {
+      const encryptionKey = new kms.EncryptionKey(this, 'ArtifactsBucketEncryptionKey', {
+        retain: false,
+      });
       propsBucket = new s3.Bucket(this, 'ArtifactsBucket', {
-        removalPolicy: cdk.RemovalPolicy.Orphan
+        removalPolicy: cdk.RemovalPolicy.Destroy,
```

Comment:
Are we sure? If the bucket is non-empty, the stack destruction operation will fail.

Reply:
I imagine this change is in response to a lot of tests cluttering up your account? :)

Maybe it should be a parameter that we can set to `Destroy` in integ tests.

Reply:
You're correct Rico. I've extracted this to a separate PR (#2241), and reverted the `removalPolicy` change (so it's still `Orphan`, as before).

Same for the KMS key (it's also retained).